cmd/minikube/cmd/cp.go

```diff
@@ -83,6 +85,25 @@ Example Command : "minikube cp a.txt /home/docker/b.txt" +
 func init() {
 }
 
+// destinationPath sets the src filename as dst filename
+// when the dst file name is not provided and ends with a `/`.
+// Otherwise this function is a no-op and returns the passed dst.
+func destinationPath(dst, src string) string {
+	if len(dst) == 0 {
+		return dst
+	}
+	if len(src) == 0 {
+		return dst
+	}
+	if !strings.HasSuffix(dst, "/") {
+		return dst
+	}
+
+	srcPaths := strings.Split(src, "/")
+	srcFileName := srcPaths[len(srcPaths)-1]
+	return dst + srcFileName
```

Comment:
No, I made the change to use `filepath` package.

Reply:
so we wanna make sure, the file path on minikube VM is linux style and on the user machine is based on their OS

Reply:
@msharran, perhaps it's easier if you use eg,

`dstDir, dstFile := filepath.Split(dst)`

and if dstFile is empty (ie, dst is ending with a separator => a directory), then just use filepath.Join(dstDir, srcFile), whereas you can extract srcFile from src in the same way as above

[filepath.Split()](https://pkg.go.dev/path/filepath#Split) and [filepath.Join()](https://pkg.go.dev/path/filepath#Join) will do the most work for you (including handling of os-specific separators):
> Split splits path immediately following the final Separator, separating it into a directory and file name component

> Join joins any number of path elements into a single path, separating them with an OS specific Separator

Reply:
btw, based on the problem described in the issue:

> I1109 16:06:40.176717   39834 ssh_runner.go:356] copy: skipping /tmp/tmp (exists)

i think that the original problem might be in https://github.com/kubernetes/minikube/blob/366d2e0733e9a20023e18be345f65bd4b3ea2376/cmd/minikube/cmd/cp.go#L167
namely, pt.Dir(dst.path) will return `all but the last element of path` for targetDir, and pt.Base(dst.path) will return `the last element of path. Trailing path separators are removed before extracting the last element` for the targetName, so for given destination of '/tmp/' it will compose to "/tmp/tmp" as shown in the issue/log

Reply:
it would probably be a good idea to also add this/similar test case in cmd/minikube/cmd/cp_test.go

Reply:
@prezha sure will check and add that

Reply:
> namely, pt.Dir(dst.path) will return all but the last element of path for targetDir, and pt.Base(dst.path) will return the last element of path. Trailing path separators are removed before extracting the last element for the targetName, so for given destination of '/tmp/' it will compose to "/tmp/tmp" as shown in the issue/log

@prezha this entire function has no tests. Do you want me to add test for `copyableFile` func or for the one that I am writing (the one that modifies dst path)?

Reply:
I have added test for `destinationPath`. That should solve the case above